pkg/api/handlers/utils/containers.go

```diff
@@ -202,12 +204,35 @@ func waitRemoved(ctrWait containerWaitFn) (int32, error) {
 	return code, err
 }
 
-func waitNextExit(ctrWait containerWaitFn) (int32, error) {
-	_, err := ctrWait(define.ContainerStateRunning)
-	if err != nil {
+func waitNextExit(ctx context.Context, containerName string) (int32, error) {
+	runtime := ctx.Value("runtime").(*libpod.Runtime)
+	var containerEngine entities.ContainerEngine = &abi.ContainerEngine{Libpod: runtime}
+	eventChannel := make(chan *events.Event)
+	errChannel := make(chan error)
```

Comment:
Should both channels be closed in a deferred statement?

Reply:
AFAIK channels don't need to be closed. It's sometime useful (e.g. to be able to end `for` loop over channel), but that's not the case here.